### PR TITLE
Globalization

### PR DESCRIPTION
**Pull request by richardpringle**
@0candy, please review.

### CODE REVIEW

**Comment by richardpringle** [comment]
Will squash commits after review.

File: lib/providers/filesystem/index.js [hunk 1]
```diff
@@ -27,21 +30,21 @@ function FileSystemProvider(options) {
   this.root = options.root;
   var exists = fs.existsSync(this.root);
   if (!exists) {
-    throw new Error('FileSystemProvider: Path does not exist: ' + this.root);
+    throw new Error(g.f('FileSystemProvider: Path does not exist: %s', this.root));
   }
```
Comment (0candy) [inline lib/providers/filesystem/index.js]: Should add {{FileSystemProvider}} for strings that don't need to be translated.

File: lib/providers/filesystem/index.js [hunk 2]
```diff
@@ -27,21 +30,21 @@ function FileSystemProvider(options) {
...
   var stat = fs.statSync(this.root);
   if (!stat.isDirectory()) {
-    throw new Error('FileSystemProvider: Invalid directory: ' + this.root);
+    throw new Error(g.f('FileSystemProvider: Invalid directory: %s', this.root));
```
Comment (0candy) [inline lib/providers/filesystem/index.js]: same as above

File: lib/providers/filesystem/index.js [hunk 3]
```diff
@@ -50,9 +53,9 @@ function validateName(name, cb) {
     return true;
   } else {
     cb && process.nextTick(cb.bind(null,
-      new Error('FileSystemProvider: Invalid name: ' + name)));
+      new Error(g.f('FileSystemProvider: Invalid name: %s', name))));
```
Comment (0candy) [inline lib/providers/filesystem/index.js]: save as above

File: lib/providers/filesystem/index.js [hunk 4]
```diff
@@ -50,9 +53,9 @@ function validateName(name, cb) {
...
     if (!cb) {
-      console.error('FileSystemProvider: Invalid name: ', name);
+      console.error(g.f('FileSystemProvider: Invalid name: ', name));
     }
```
Comment (0candy) [inline lib/providers/filesystem/index.js]: same as above

File: lib/storage-handler.js [hunk 5]
```diff
@@ -43,7 +46,7 @@ exports.upload = function(provider, req, res, options, cb) {
       part.on('data', function(buffer) {
         self._fieldsSize += buffer.length;
         if (self._fieldsSize > self.maxFieldsSize) {
-          self._error(new Error('maxFieldsSize exceeded, received ' + self._fieldsSize + ' bytes of field data'));
+          self._error(new Error(g.f('maxFieldsSize exceeded, received %s bytes of field data', self._fieldsSize)));
```
Comment (0candy) [inline lib/storage-handler.js]: Should add brackets {{maxFieldsSize}}

File: lib/storage-handler.js [hunk 6]
```diff
@@ -88,7 +91,7 @@ exports.upload = function(provider, req, res, options, cb) {
       }
       if (Array.isArray(allowedContentTypes) && allowedContentTypes.length !== 0) {
         if (allowedContentTypes.indexOf(file.type) === -1) {
-          self._error(new Error('contentType "' + file.type + '" is not allowed (Must be in [' + allowedContentTypes.join(', ') + '])'));
+          self._error(new Error(('contentType "%s" is not allowed (Must be in [%s])', file.type, allowedContentTypes.join(', '))));
```
Comment (0candy) [inline lib/storage-handler.js]: Should add brackets {{contentType}}

File: lib/storage-handler.js [hunk 7]
```diff
@@ -155,7 +158,7 @@ exports.upload = function(provider, req, res, options, cb) {
           // We are missing some way to tell the provider to cancel upload/multipart upload of the current file.
           // - s3-upload-stream doesn't provide a way to do this in it's public interface
           // - We could call provider.delete file but it would not delete multipart data 
-          self._error(new Error('maxFileSize exceeded, received ' + fileSize + ' bytes of field data (max is ' + maxFileSize + ')'));
+          self._error(new Error(g.f('maxFileSize exceeded, received %s bytes of field data (max is %s)', fileSize, maxFileSize)));
           return;
```
Comment (0candy) [inline lib/storage-handler.js]: Should add brackets {{maxFileSize}}

**Comment by 0candy** [comment]
You have to run slt-globalize -e again for intl/en/messages.json to get updated with the brackets change.

**Comment by rmg** [comment]
@slnode test please

**Comment by richardpringle** [comment]
@rmg,

`npm test` is currently failing out of the box. I am currently investigating the cause.

**Comment by richardpringle** [comment]
@0candy, please review round 2

**Comment by 0candy** [comment]
Code change itself LGTM.

**Comment by richardpringle** [comment]
@slnode test please

**Pull request merged**